src/raster/rasterband.rs

```diff
@@ -600,3 +612,51 @@ impl ColorInterpretation {
         _string(rv)
     }
 }
+
+pub struct ColorTable<'a> {
+    c_color_table: GDALColorTableH,
+    raster_band: &'a RasterBand<'a>,
+}
+
+impl<'a> ColorTable<'a> {
+    pub fn raster_band(&self) -> &RasterBand {
+        self.raster_band
+    }
+
+    /// How the values of this color table are interpreted.
+    pub fn palette_interpretation(&self) -> GDALPaletteInterp::Type {
+        unsafe { gdal_sys::GDALGetPaletteInterpretation(self.c_color_table) }
+    }
+
+    /// Get the number of color entries in this color table.
+    pub fn entry_count(&self) -> i32 {
+        unsafe { gdal_sys::GDALGetColorEntryCount(self.c_color_table) }
+    }
+
+    /// Get a color entry.
+    pub fn entry(&self, index: i32) -> Option<GDALColorEntry> {
+        unsafe {
+            let c_color_entry = gdal_sys::GDALGetColorEntry(self.c_color_table, index);
+            if c_color_entry.is_null() {
+                return None;
+            }
+            Some(*c_color_entry)
+        }
+    }
+
+    /// Get a color entry as RGB.
+    pub fn entry_as_rgb(&self, index: i32) -> Option<GDALColorEntry> {
+        let mut color_entry = GDALColorEntry {
+            c1: 0,
+            c2: 0,
+            c3: 0,
+            c4: 0,
+        };
+        if unsafe { gdal_sys::GDALGetColorEntryAsRGB(self.c_color_table, index, &mut color_entry) }
+            == 0
+        {
+            return None;
+        }
+        Some(color_entry)
+    }
+}
```

Comment:
Could you create a test that uses each of these methods?
Otherwise, we don't know that they work.

As far as I see there is no need to free any value wrt. leaking.

Reply:
Can you generate some small test data, for instance with the GDAL CLI?

Reply:
Is there someplace specific that I should put the test file (a small geo-tiff, 22k) or is the raster folder okay?

Reply:
Done.